src/genericsite/models.py

```diff
@@ -80,6 +82,136 @@ def For(cls, site: T.Union[Site, int]) -> models.Manager:
+    name of the new field where their audio/video/attachment is stored.
+
+    If `mime_type` is not populated, it will be guessed when the instance is saved,
+    based on the file name stored in the `content_field` (using `mimetypes.guess_type`).
+    """
+
+    class Meta:
+        abstract = True
+
+    content_field = "image_file"
+
+    title = models.CharField(_("title"), max_length=255, unique=True)
+    description = models.CharField(
+        verbose_name=_("description"), max_length=255, blank=True
+    )
+    image_file = ThumbnailerImageField(
+        _("image file"),
+        width_field="image_width",
+        height_field="image_height",
+        max_length=255,
+    )
+    image_width = models.IntegerField(_("image_width"), blank=True)
+    image_height = models.IntegerField(_("image_height"), blank=True)
+    site = models.ForeignKey(Site, on_delete=models.CASCADE, default=1)
+
+    tags = TaggableManager(blank=True)
+
+    # Open Graph has `type`, Schema.org calls it `encodingFormat`, both recommend MIME
+    # type as the value
+    mime_type = models.CharField(
+        _("MIME type"), max_length=255, db_index=True, blank=True
+    )
+    copyright_holder = models.CharField(
+        _("copyright holder"), max_length=255, blank=True
+    )
+    custom_copyright_notice = HTMLField(
+        _("custom copyright notice"),
+        blank=True,
+        help_text=_(
+            r"include a pair of curly braces {} where you want the year inserted"
+        ),
+    )
+    uploaded_dt = models.DateTimeField(_("when uploaded"), auto_now_add=True)
+    created_dt = models.DateTimeField(_("when taken"), blank=True, null=True)
+
+    @property
+    def copyright_year(self):
+        if self.created_dt:
+            return self.created_dt.year
+        else:
+            return self.uploaded_dt.year
+
+    @property
+    def copyright_notice(self):
+        conf = apps.get_app_config("genericsite")
+        var = self.site.vars
+        if self.custom_copyright_notice:
+            return format_html(self.custom_copyright_notice, self.copyright_year)
+        elif notice := var.get_value("copyright_notice"):
+            return format_html(notice, self.copyright_year)
+        else:
+            holder = var.get_value("copyright_holder", self.site.name)
+            return format_html(conf.fallback_copyright, self.copyright_year, holder)
+
+    def __str__(self):
+        return self.title
+
+    @property
+    def _base_url(self):
+        protocol = "http" if settings.DEBUG else "https"
+        return f"{protocol}://{self.site.domain}"
+
+    @property
+    def url(self):
+        path = getattr(self, self.content_field).url
+        protocol = "http" if settings.DEBUG else "https"
+        return f"{self._base_url}{path}"
+
+    def save(self, *args, **kwargs):
+        if not self.mime_type:
+            self.mime_type, _ = mimetypes.guess_type(
+                getattr(self, self.content_field).name, strict=False
```

Comment:
Taking your questions in order: 
1. If it returns None, then the field will be NULL in the database, which is fine, because the field is nullable and not really used for anything important (or at all, just now).
2. Django's `ImageField` (and the `ThumbnailerImageField` subclass) checks the file extension to ensure it is something that PIL knows how to process and will raise a `ValidationError` if not.
3. If the Thumbnailer gets something it doesn't know how to thumbnail, it will raise an exception, but it doesn't know or care about this `mime_type` field, it looks at the file itself.